lattices/src/seq.rs

```diff
@@ -0,0 +1,162 @@
+use std::cmp::Ordering::{self, *};
+
+use cc_traits::Iter;
+
+use crate::{ConvertFrom, LatticeOrd, Merge};
+
+/// Sequence compound lattice.
+///
+/// Contains any number of `Lat` sub-lattices. Sub-lattices are indexed starting at zero, merging
+/// combines corresponding sub-lattices and keeps any excess.
+///
+/// Similar to [`MapUnion<<usize, Lat>>`](super::map_union::MapUnion) but requires the key indices
+/// start at `0` and have no gaps.
+#[derive(Clone, Debug, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Seq<Lat> {
+    seq: Vec<Lat>,
+}
+
+impl<Lat> Seq<Lat> {
+    /// Create a new `Seq` from a `Vec` of `Lat` instances.
+    pub fn new(seq: Vec<Lat>) -> Self {
+        Self { seq }
+    }
+
+    /// Create a new `Seq` from an `Into<Vec<Lat>>`.
+    pub fn new_from(seq: impl Into<Vec<Lat>>) -> Self {
+        Self::new(seq.into())
+    }
+}
+
+impl<Lat> Default for Seq<Lat> {
+    fn default() -> Self {
+        Self {
+            seq: Default::default(),
+        }
+    }
+}
+
+impl<LatSelf, LatOther> Merge<Seq<LatOther>> for Seq<LatSelf>
+where
+    LatSelf: Merge<LatOther> + ConvertFrom<LatOther>,
+{
+    fn merge(&mut self, other: Seq<LatOther>) -> bool {
```

Comment:
does this work:

```rust
fn merge(&mut self, other: Seq<LatOther>) -> bool {
    let mut changed = false;

    self.seq.extend(other.seq.drain(self.seq.len()..));

    for (self_val, other_val) in self.seq.iter_mut().zip(other.seq.into_iter()) {
        changed |= self_val.merge(other_val);
    }

    changed
}
```

Reply:
not quite, `drain` panics if you put a range off the end of the vector, and also changed needs to be set for the extend

this adjusted version works though:
```rust
    fn merge(&mut self, mut other: Seq<LatOther>) -> bool {
        let mut changed = false;
        // Extend `self` if `other` is longer.
        if self.seq.len() < other.seq.len() {
            self.seq
                .extend(other.seq.drain(self.seq.len()..).map(LatSelf::from));
            changed = true;
        }
        // Merge intersecting indices.
        for (self_val, other_val) in self.seq.iter_mut().zip(other.seq.into_iter()) {
            changed |= self_val.merge(other_val);
        }
        changed
    }
```